airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumRecordIterator.java

```diff
@@ -80,9 +82,12 @@ protected ChangeEvent<String, String> computeNext() {
       // if within the timeout, the consumer could not get a record, it is time to tell the producer to
       // shutdown.
       if (next == null) {
-        LOGGER.info("Closing cause next is returned as null");
-        requestClose();
+        if ((!receivedFirstRecord || hasSnapshotFinished || maxInstanceOfNoRecordsFound >= 10) && !signalledClose) {
+          LOGGER.info("Closing cause next is returned as null");
```

Comment:
Can you add a comment or add a better log message here to reflect the new logic?